src/server/saved_objects/migrations/core/elastic_index.ts

```diff
@@ -297,6 +308,23 @@ async function assertIsSupportedIndex(indexInfo: FullIndexInfo) {
   return indexInfo;
 }
 
+/**
+ * Provides protection against reading/re-indexing against an index with missing
+ * shards which could result in data loss. This shouldn't be common, as the Saved
+ * Object indices should only ever have a single shard. This is more to handle
+ * instances where customers manually expand the shards of an index.
+ */
+function assertResponseIncludeAllShards({ _shards }: { _shards: ShardsInfo }) {
+  const failed = _shards.total - _shards.successful;
```

Comment:
I am making an assumption that `_shards` is always returned by a search request, which I believe is the case.

Reply:
thoughts on returning early if response.shards is undefined?  i thought a more on this and if there's an error  it may be undefined e.g. searching on an index that doesn't exist.  if r.shards doesn't exist we can prob assume a future or prior error handler will catch it

Reply:
i think the client will throw, but that's what we were assuming with reindex.  in practice i think it's as is fine but i don't think it'll hurt